api/test/e2e/consumer_unusual_test.go

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tidwall/gjson"
+)
+
+//case 4: delete consumer(id: not_found)
+func TestConsumer_delete_notexit_consumer(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc:     "delete notexit consumer",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/consumers/notexit",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusNotFound,
+		},
+	}
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+}
+
+//case 8: update consumer, check if updatetime is updated
+func TestConsumer_create_consumer_with_createtime_updatetime(t *testing.T) {
+	//create consumer, save the result_A ( create_time and update_time )
+	data := `{
+		"username":"case_8",
+		"desc": "new consumer"
+   }`
+	request, _ := http.NewRequest("PUT", "http://127.0.0.1:8080/apisix/admin/consumers", strings.NewReader(data))
```

Comment:
Is it better to extract public variables? such as `BASE_URL = http://127.0.0.1:8080/apisix/admin/consumers`